components/heatmap/HeatMap.vue

```diff
@@ -235,7 +235,7 @@ export default {
     passing_plate_colors: function () {
       return this.well_values[this.display_option].data.map((well) => {
         const total = well.slice(-MIN_NUM_DATAPOINTS_FOR_MEAN).reduce((a, b) => a + b, 0);
-        const mean = (total / MIN_NUM_DATAPOINTS_FOR_MEAN).toFixed(3);
+        const mean = (total / well.slice(-MIN_NUM_DATAPOINTS_FOR_MEAN).length).toFixed(3);
```

Comment:
Should pull out `well.slice(-MIN_NUM_DATAPOINTS_FOR_MEAN)` into its own variable here so it's not duplicated